### ISSUE TITLE
"Improve rumble performance and stability" commit seems to do the opposite

### ISSUE THREAD

**Issue by v1993** [DanielOgorchock/linux#22]
This have been observed both by me and [some other people](https://github.com/nicman23/dkms-hid-nintendo/issues/33) that last change actually causes Bluetooth to drop connection more often.

**Comment by v1993** [comment]
More testing reveals that issues happen on both versions, but they are different. Old method gives more disconnects but accel/gyro drifts less while new is less likely to disconnect but accel/gyro drifts more during rumble.

**Comment by nicman23** [comment]
i am thinking of disabling support for it in the dkms package tbh.

**Comment by v1993** [comment]
@nicman23 Do you mean rumble or gyro? I think that it would be fine to keep both, it's usually possible to turn rumble off in games and it works OK in scenarios where rumble is not too heavy.

**Comment by nicman23** [comment]
i mean rumble it is not stable for bluetooth.

**Comment by DanielOgorchock** [comment]
Yeah rumble over bluetooth hasn't been stable with a couple adapters I've tried. I do believe the patch is an improvement though. It actually results in less traffic from rumble and more closely reflects what the Switch does.

IIRC using the driver with the switchroot project on an actual console yielded more reliable results for me. I wonder if it's due to the bluetooth adapters in question.

**Comment by DanielOgorchock** [comment]
I think my latest patchset improves BT rumble reliability by a large margin. Guess we'll have to wait and see as people start using the improvements
